apis/r/R/SOMASparseNDArray.R

```diff
@@ -114,7 +114,7 @@ SOMASparseNDArray <- R6::R6Class(
     #' @param iterated Option boolean indicated whether data is read in call (when
     #' `FALSE`, the default value) or in several iterated steps.
     #' @param log_level Optional logging level with default value of `"warn"`.
-    #' @return An [`arrow::Table`].
+    #' @return arrow::\link[arrow]{Table} or \link{TableReadIter}
```

Comment:
As above: by group discussion and decision some months ago we decided to replace `read_foo()` with separate `read()` chained with `foo()`. See #656 and #663 for context.

This needs to be just `read`, always returning Arrow table, and its return value needs conversation methods to convert to other formats.